stream.php

```diff
@@ -29,361 +29,23 @@
-
-		ksort( $allowed_html );
-
-		foreach ( self::$notices as $notice ) {
-			$class_name   = empty( $notice['is_error'] ) ? 'updated' : 'error';
-			$html_message = sprintf( '<div class="%s">%s</div>', esc_attr( $class_name ), wpautop( $notice['message'] ) );
-
-			echo wp_kses( $html_message, $allowed_html );
-		}
-	}
-
-	/**
-	 * Displays an HTML comment in the frontend head to indicate that Stream is activated,
-	 * and which version of Stream is currently in use.
-	 *
-	 * @action wp_head
-	 *
-	 * @return string|void An HTML comment, or nothing if the value is filtered out.
-	 */
-	public function frontend_indicator() {
-		$comment = sprintf( 'Stream WordPress user activity plugin v%s', esc_html( self::VERSION ) ); // Localization not needed
-
-		/**
-		 * Filter allows the HTML output of the frontend indicator comment
-		 * to be altered or removed, if desired.
-		 *
-		 * @since 1.4.5
-		 *
-		 * @return string  The content of the HTML comment
-		 */
-		$comment = apply_filters( 'wp_stream_frontend_indicator', $comment );
-
-		if ( ! empty( $comment ) ) {
-			echo sprintf( "<!-- %s -->\n", esc_html( $comment ) ); // xss ok
-		}
-	}
-
-	/**
-	 * Version of plugin_dir_url() which works for plugins installed in the plugins directory,
-	 * and for plugins bundled with themes.
-	 *
-	 * @throws \Exception
-	 *
-	 * @return array
-	 */
-	private function locate_plugin() {
-		$reflection = new \ReflectionObject( $this );
-		$file_name  = $reflection->getFileName();
-
-		if ( '/' !== \DIRECTORY_SEPARATOR ) {
-			$file_name = str_replace( \DIRECTORY_SEPARATOR, '/', $file_name ); // Windows compat
-		}
-
-		$plugin_dir = preg_replace( '#(.*plugins[^/]*/[^/]+)(/.*)?#', '$1', $file_name, 1, $count );
-
-		if ( 0 === $count ) {
-			throw new \Exception( "Class not located within a directory tree containing 'plugins': $file_name" );
-		}
-
-		// Make sure that we can reliably get the relative path inside of the content directory
-		$content_dir = trailingslashit( WP_CONTENT_DIR );
-
-		if ( '/' !== \DIRECTORY_SEPARATOR ) {
-			$content_dir = str_replace( \DIRECTORY_SEPARATOR, '/', $content_dir ); // Windows compat
-		}
-
-		if ( 0 !== strpos( $plugin_dir, $content_dir ) ) {
-			throw new \Exception( 'Plugin dir is not inside of WP_CONTENT_DIR' );
-		}
-
-		$content_sub_path = substr( $plugin_dir, strlen( $content_dir ) );
-		$dir_url          = content_url( trailingslashit( $content_sub_path ) );
-		$dir_path         = trailingslashit( $plugin_dir );
-		$dir_basename     = basename( $plugin_dir );
-		$plugin_basename  = trailingslashit( $dir_basename ) . basename( __FILE__ );
-
-		return compact( 'dir_url', 'dir_path', 'dir_basename', 'plugin_basename' );
-	}
-
-}
-
-if ( ! WP_Stream::is_valid_php_version() ) {
-	WP_Stream::fail_php_version();
+if ( ! version_compare( PHP_VERSION, '5.3', '>=' ) ) {
+	load_plugin_textdomain( 'stream', false, dirname( plugin_basename( __FILE__ ) ) . '/languages/' );
+	add_action( 'shutdown', function() {
```

Comment:
@lukecarbis You can't use an anonymous function here because that would require PHP 5.3 :wink: